src/osg/ApplicationUsage.cpp

```diff
@@ -286,7 +286,6 @@ void ApplicationUsage::write(std::ostream& output, unsigned int type, unsigned i
         if (needspace) output << std::endl;
         output << "Keyboard and Mouse Bindings:"<<std::endl;
         write(output,getKeyboardMouseBindings(),widthOfOutput);
-        needspace = true;
```

Comment:
Same issue, cpphcheck might be technically correct the needspace is redundant but absolutely the wrong thing to do from a living and breathing body of C++ code that is open to revisions over time - if one adds another one of these blocks but not add the needSpace you'd get a bug.  Far better to leave the redundant assignment.  The consequence performance wise is utterly irrelevant as it's not an inner loop, and only ever called once in very special circumstances.

Here again wasting our time of silly false positives.